### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Subobject.Basic

### PR DESCRIPTION
**Pull request by joelriou**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Subobject/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,704 @@
...
+Currently we describe `pullback`, `map`, etc., as functors.
+It may be better to just say that they are monotone functions,
+and even avoid using categorical language entirely when describing `Subobject X`.
+(It's worth keeping this in mind in future use; it should be a relatively easy change here
+if it looks preferable.)
+
+### Relation to pseudoelements
+
+There is a separate development of pseudoelements in `CategoryTheory.Abelian.Pseudoelements`,
+as a quotient (but not by isomorphism) of `Pver X`.
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Subobject/Basic.lean:66]:
```suggestion
as a quotient (but not by isomorphism) of `Over X`.
```

File: Mathlib/CategoryTheory/Subobject/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,704 @@
...
+theorem mk_eq_mk_of_comm {B A₁ A₂ : C} (f : A₁ ⟶ B) (g : A₂ ⟶ B) [Mono f] [Mono g] (i : A₁ ≅ A₂)
+    (w : i.hom ≫ g = f) : mk f = mk g :=
+  eq_mk_of_comm _ ((underlyingIso f).trans i) <| by simp [w]
+#align category_theory.subobject.mk_eq_mk_of_comm CategoryTheory.Subobject.mk_eq_mk_of_comm
+
+-- We make `X` and `Y` explicit arguments here so that when `ofLe` appears in goal statements
+-- it is possible to see its source and target
+-- (`h` will just display as `_`, because it is in `Prop`).
+/-- An inequality of subobjects is witnessed by some morphism between the corresponding objects. -/
+def ofLe {B : C} (X Y : Subobject B) (h : X ≤ Y) : (X : C) ⟶ (Y : C) :=
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Subobject/Basic.lean:319]:
```suggestion
def ofLe {B : C} (X Y : Subobject B) (h : X ≤ Y) : (X : C) ⟶ (Y : C) :=
```
I think we have decided to keep `LE` capitalised even when it is part of a camelCase identifier, although this is fair from uniformly followed. I asked at https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Naming.20convention/near/350418574 for clarification on the consensus here.

I'll delegate this PR shortly, feel free to merge either with or without addressing this --- it's a widespread enough inconsistency there's no need to hold up this one in particular.

**Comment by semorrison** [comment]
bors d+

File: Mathlib/CategoryTheory/Subobject/Basic.lean (hunk 2 above)
Comment (joelriou) [reply Mathlib/CategoryTheory/Subobject/Basic.lean:319]: Indeed, I was not sure. I shall change `Le` to `LE` later today.

**Pull request closed without merging**